reframe/core/schedulers/slurm.py

```diff
@@ -193,6 +197,17 @@ def _get_reservation_nodes(self):
         node_descriptions = completed.stdout.splitlines()
         return (SlurmNode(descr) for descr in node_descriptions)
 
+    def _get_excluded_nodes(self):
+        command = 'scontrol show -o -a %s' % self.sched_exclude_nodelist
+        completed = os_ext.run_command(command, check=True)
+
+        node_descriptions_output = completed.stdout()
+        if node_descriptions == '':
+            raise JobError("could not extract the node description for "
+                           "excluded nodes '%s'" % self.sched_exclude_nodelist)
```

Comment:
What does the empty output mean? If anything happens during the `scontrol show` an exception will be thrown from the `run_command` because you have the `check=True`. I suggest catching the `SpawnedProcessError` and reraising it as a `JobError` if you like:

```python
try:
    os_ext.run_command(...)
except SpawnedProcessError as e:
    raise JobError('could not retrieve the node description of nodes: %s' self.sched_exclude_nodelist) from e
```